src/Filter/ChoiceFilter.php

```diff
@@ -54,8 +66,19 @@ public function getRenderSettings()
         ]];
     }
 
-    private function filterWithMultipleValues(ProxyQueryInterface $queryBuilder, string $alias, string $field, array $data = []): void
+    private function filterWithMultipleValues(BaseProxyQueryInterface $query, string $alias, string $field, array $data = []): void
     {
+        /* NEXT_MAJOR: Remove this deprecation and update the typehint */
+        if (!$query instanceof ProxyQueryInterface) {
```

Comment:
It's not necessary to trigger a deprecation in this private method, just the `NEXT_MAJOR` comment.